hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ContainerImporter.java

```diff
@@ -65,9 +63,7 @@ public ContainerImporter(ConfigurationSource conf, ContainerSet containerSet,
     this.controller = controller;
     this.volumeSet = volumeSet;
     try {
-      volumeChoosingPolicy = conf.getClass(
-          HDDS_DATANODE_VOLUME_CHOOSING_POLICY, RoundRobinVolumeChoosingPolicy
-              .class, VolumeChoosingPolicy.class).newInstance();
+      volumeChoosingPolicy = VolumeChoosingPolicyFactory.getPolicy(conf);
```

Comment:
Why is there a need for a volumeChoosingFactory here? Can't we just add the DEFAULT_VOLUME_CHOOSING_POLICY in a constant class?

Reply:
Classes inside `org.apache.hadoop.ozone.container.common.volume` can't be accessed in HDDSConfigKeys as the `common` module doesn't depend on the` container-service` module. The other way is to hardcode the class name as a string but this might pose a problem if some one changes the location of the class.

Just checked Even Container placement policy uses a factory class to get its impls.